### PR TITLE
[HTTP] Add support for 100-continue

### LINKED ISSUE #3793: [HTTP] Handle Expect header
At the moment we do not have a nice way to handle Expect header and then send back whatever the desired response. IMO, Request struct must have a native function to check if the request has Expect header and then another native function to respond with 100 continue.

### PR DESCRIPTION
**Pull request by pubudu91**
Fix #3793

### CODE REVIEW

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/httplib.bal [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package ballerina.net.http;
+
+const string CONTINUE_100 = "100-continue";
+
+@Description { value:"This function can be used to check whether the client expects a 100-continue response."}
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/ballerina/ballerina/net/http/httplib.bal]: Shall we remove "This function can be used to" part from the description? IMO, Something like "Checks whether the client expects a 100-continue response or not" will be better

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/httplib.bal [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+@Param { value:"req: A request struct" }
+@Return { value:"Returns true if the client expects a 100-continue response. If not, returns false." }
+public function <Request req> expects100Continue() (boolean) {
+    string expectHeader = req.getHeader("Expect");
+
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/ballerina/ballerina/net/http/httplib.bal]: Shall we remove this new line? Looks inconsistent with other contents in the file

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/httplib.bal [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package ballerina.net.http;
+
+const string CONTINUE_100 = "100-continue";
+
+@Description { value:"This function can be used to check whether the client expects a 100-continue response."}
+@Param { value:"req: A request struct" }
+@Return { value:"Returns true if the client expects a 100-continue response. If not, returns false." }
+public function <Request req> expects100Continue() (boolean) {
+    string expectHeader = req.getHeader("Expect");
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/ballerina/ballerina/net/http/httplib.bal]: Shall we define a const for "Expect" as well?

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/httplib.bal [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    if (expectHeader == CONTINUE_100) {
+        return true;
+    }
+
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/ballerina/ballerina/net/http/httplib.bal]: Shall we remove this new line too? Looks inconsistent with other contents in the file

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/httplib.bal (hunk 1 above)
Comment (pubudu91) [reply modules/ballerina-http/src/main/ballerina/ballerina/net/http/httplib.bal]: Ack.

File: modules/ballerina-http/src/main/ballerina/ballerina/net/http/httplib.bal (hunk 3 above)
Comment (pubudu91) [reply modules/ballerina-http/src/main/ballerina/ballerina/net/http/httplib.bal]: Ack.

**Pull request merged**